test/TypeCoercion/protocol_covariance.swift

```diff
@@ -0,0 +1,64 @@
+// RUN: %target-typecheck-verify-swift
+
+class A {}
```

Comment:
Please put this test in test/decl/protocol, and add some error cases as well.

Also you need a SILGen test and an executable test.